### PR TITLE
fix: schema for the `host` option

### PR DESCRIPTION
**Pull request by alexander-akait**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Existing

### Motivation / Use-Case

fix schema

### Breaking Changes

No

### Additional Info

Also update options for CLI

### CODE REVIEW

**Review by anshumanv — APPROVED** [review]

File: lib/options.json [hunk 1]
```diff
@@ -283,10 +283,17 @@
...
+      "link": "https://webpack.js.org/configuration/dev-server/#devserverhost",
+      "anyOf": [
+        {
+          "enum": ["local-ip", "local-ipv4", "local-ipv6"]
+        },
+        {
+          "type": "string",
+          "minLength": 1
+        }
+      ]
```
Comment (snitin315) [inline lib/options.json:296]: `schema-utils` should respect `parentSchema.link/description`, if we don't have link/description for children.

```suggestion
      "anyOf": [
        {
          "enum": ["local-ip", "local-ipv4", "local-ipv6"],
          "link": "https://webpack.js.org/configuration/dev-server/#devserverhost"
        },
        {
          "type": "string",
          "minLength": 1,
          "link": "https://webpack.js.org/configuration/dev-server/#devserverhost"
        }
      ]
```

File: test/__snapshots__/validate-options.test.js.snap.webpack4 [hunk 2]
```diff
@@ -258,23 +258,31 @@ exports[`options validate should throw an error on the "historyApiFallback" opti
 
 exports[`options validate should throw an error on the "host" option with '' value 1`] = `
 "ValidationError: Invalid options object. Dev Server has been initialized using an options object that does not match the API schema.
- - options.host should be a non-empty string.
-   -> Allows to specify a hostname to use.
-   -> Read more at https://webpack.js.org/configuration/dev-server/#devserverhost"
+ - options.host should be a non-empty string."
 `;
```
Comment (snitin315) [inline test/__snapshots__/validate-options.test.js.snap.webpack4:262]: I see the link is not showing only happening with `non-empty` error.

Comment (alexander-akait) [reply test/__snapshots__/validate-options.test.js.snap.webpack4:262]: Yep, it should be fixed in `schema-utils`

Comment (alexander-akait) [reply test/__snapshots__/validate-options.test.js.snap.webpack4:262]: I think we can merge it as is and update snapshots after fixing it in schema utils

**Pull request merged**